src/units/ability.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+	Copyright (C) 2003 - 2022
+	by David White <dave@whitevine.net>
+	Part of the Battle for Wesnoth Project https://www.wesnoth.org/
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation; either version 2 of the License, or
+	(at your option) any later version.
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY.
+
+	See the COPYING file for more details.
+*/
+
+#include "units/ability.hpp"
+
+void unit_ability_t::parse_vector(const config& abilities_cfg, std::vector<ability_ptr>& res)
```

Comment:
Is there a reason to take this by reference rather than returning it?

I would prefer returning it unless there is a good reason to do otherwise (or rather, just merge this code into `cfg_to_vector`).

I see it's called in two places beyond this file, but one of those is in a constructor and the other one clears the list first, so neither of them rely on it being a reference parameter.

Reply:
> one of those is in a constructor

It supports multiple [abilities] tag (idk why, but it was the previous behaviour) so using cfg_to_vector wouold be more complicated there

Reply:
If that's the case it looks like your refactor might have missed it, because I didn't see anywhere that called it more than once, eg on a loop.

Reply:
In unit::init which is i think the only place that uses it.